tests/backup/backup_restore_basic_test.go

```diff
@@ -1104,18 +1146,21 @@ var _ = Describe("{AllNSBackupWithIncludeNewNSOption}", func() {
 			dash.VerifySafely(err, nil, fmt.Sprintf("Verification of deleting backup schedule - %s", scheduleName))
 		}
 		log.Infof("Deleting backup schedule policy")
-		policyList := []string{periodicSchedulePolicyName}
+		policyList := []string{schedulePolicyName}
 		err = Inst().Backup.DeleteBackupSchedulePolicy(orgID, policyList)
 		dash.VerifySafely(err, nil, fmt.Sprintf("Deleting backup schedule policies %s ", policyList))
 		err = DeleteRestore(restoreName, orgID, ctx)
 		dash.VerifySafely(err, nil, fmt.Sprintf("Deleting restore [%s]", restoreName))
 
 		opts := make(map[string]bool)
 		opts[SkipClusterScopedObjects] = true
-		log.InfoD("Deleting deployed namespaces - %v", appNamespaces)
-		ValidateAndDestroy(contexts, opts)
+		allAppNamespaces := append(appNamespaces, newAppNamespaces...)
+		log.InfoD("Deleting deployed namespaces - %v", allAppNamespaces)
+		ValidateAndDestroy(allContexts, opts)
 
 		CleanupCloudSettingsAndClusters(backupLocationMap, cloudCredName, cloudCredUID, ctx)
+		err = SetSourceKubeConfig()
```

Comment:
Can we add `defer SetSourceKubeConfig()` and put this as the first line in `JustAfterEach `

Reply:
I am skeptical about using defer for a function that throws an error.